frontend/simple/model/contracts/group.js

```diff
@@ -109,63 +109,81 @@ export default DefineContract({
   },
   'GroupProposal': {
     constants: {
-      TypeInvitation: 'invitationProposal',
-      TypeRemoval: 'removalProposal',
-      TypeChange: 'changeProposal'
+      // what the proposal does
+      // if updating group field, this should be the field name to update
+      DoInvitation: 'invitation',
+      DoRemoval: 'removal',
+      DoIncome: 'incomeProvided',
+      DoChangeThreshold: 'changeThreshold',
+      DoApprovalThreshold: 'memberApprovalThreshold',
+      DoRemovalThreshold: 'memberRemovalThreshold',
+      // how the proposal vote gets decided
+      // TODO better place for these? + other types?
+      RuleTypeThreshold: 'ruleTypeThreshold'
     },
     validate: objectOf({
-      type: string,
-      threshold: number,
-      candidate: string,
+      whoProposed: string,
+      doWhat: string,
+      toWhat: string,
+      whenProposed: string,
+      voteRule: objectOf({
```

Comment:
The `votingRule` should be a string, not an object (see whiteboard). It should not include any other information like threshold, etc.

Reply:
You can name it `votingRuleType`

Reply:
oh, this brings up an interesting question! what happens when a rule changes while a proposal is open? should it be evaluated based on the rule at the proposal creation's moment, or should it be reevaluated at every vote, and should it be reevaluated when the rule changes?

Reply:
I think it makes sense to evaluate both:

- On every vote
- Re-evaluate all open proposals that use `votingRuleType` whenever a voting rule of that type changes

Reply:
If we re-evaluate open proposals mid-vote, we should notify those who've already voted because it may change how some wish to vote.

Reply:
@dotmacro for threshold-based voting that shouldn't matter. but if a clear use case is identified then an open issue should be created

Reply:
Perhaps I misunderstand the voting logic. Are we closing proposals once the threshold is reached, or only after everyone has voted?

Reply:
Once the voting rule decides the vote has passed

Reply:
Ah, then I misunderstood. Re-evaluating shouldn't matter, and makes sense 👍